weldx/measurement.py

```diff
@@ -37,71 +28,705 @@ class Signal:
+            added to the returned signal
+
+        Returns
+        -------
+        Signal :
+            The resulting signal
+
+        """
+        return Signal(
+            signal_type=cls._determine_output_signal_type(
+                transformation.type_transformation, input_signal.signal_type
+            ),
+            unit=cls._determine_output_signal_unit(
+                transformation.func, input_signal.unit
+            ),
+            data=data,
+        )
+
+    @staticmethod
+    def _determine_output_signal_type(type_transformation: str, input_type: str) -> str:
+        """Determine the type of a transformations' output signal.
+
+        Parameters
+        ----------
+        type_transformation :
+            The string describing the type transformation
+        input_type :
+            The type of the input signal
+
+        Returns
+        -------
+        str :
+            The type of the output signal
+
+        """
+        if type_transformation is not None:
+            lookup = dict(A="analog", D="digital")
+            exp_input_type = lookup.get(type_transformation[0])
+
+            if input_type != exp_input_type:
+                raise ValueError(
+                    f"The transformation expects an {exp_input_type} as input signal, "
+                    f"but the current signal is {input_type}."
+                )
+            return lookup.get(type_transformation[1])
+
+        return input_type
 
     @staticmethod
-    def _add_node(
-        node: str,
-        parent_node: str,
-        node_label: str,
-        position: Tuple[float, float],
-        container: Tuple["DiGraph", List, Dict, Dict],
-    ):  # pragma: no cover
-        """Add a new node to several containers.
-
-        This is a helper for the plot function.
-
-        Parameters
-        ----------
-        node :
-            Name of the new node
-        parent_node :
-            Name of the parent node
-        node_label :
-            Displayed name of the node
-        position :
-            Position of the node
-        container :
-            Tuple of containers that should be updated.
-
-        """
-        [graph, node_list, labels, positions] = container
-        graph.add_node(node)
-        node_list.append(node)
-        labels[node] = node_label
-        positions[node] = position
-        if parent_node is not None:
-            graph.add_edge(parent_node, node)
+    def _determine_output_signal_unit(
+        func: "MathematicalExpression", input_unit: str
+    ) -> str:
+        """Determine the unit of a transformations' output signal.
+
+        Parameters
+        ----------
+        func :
+            The function describing the transformation
+        input_unit :
+            The unit of the input signal
+
+        Returns
+        -------
+        str:
+            Unit of the transformations' output signal
+
+        """
+        from weldx import Q_
```

Comment:
any reason this can not be a top level import?

Reply:
I am not sure about this particular import, but there are ring dependency issues. Can test it for this case and report back

Reply:
Confirmed the circular dependency

Reply:
that's weird since this come straight from `constants` ..?

Reply:
Managed to get it working by importing directly from `weldx.constants` as in the `__init__.py`. So importing `weldx` was the problem. Got to remember this in the future